File: graphs/bfs.cpp
#include <bits/stdc++.h>
using namespace std;
vector<int> v[100];
bool visited[100];
int level;
queue<int> q;
int main(){
    int a, b, n, e, s, i, j, k, l;
    cin >> n >> e >> s;
    for (i = 1; i <= e; i++){
        cin >> a >> b;
        v[a].push_back(b);
        v[b].push_back(a);
    }
    for (i = 0; i <= n; i++){
        visited[i] = false;
    }
    q.push(s);
    visited[s] = true;
    level = 0;
    int siz = 0;
    while (!q.empty()){
        printf("Level = %d\n", level);
        siz = q.size();
        while (siz--){
            int f = q.front();
            q.pop();
            printf("%d\n", f);
            for (i = 0; i < v[f].size(); i++){
                if (visited[v[f][i]] == false){
                    visited[v[f][i]] = true;
                    q.push(v[f][i]);
                }
            }
        }
        level++;
    }
}

File: graphs/bfs.py
from collections import defaultdict

graph = defaultdict(list)
n, e = map(int, input().split())
s = int(input())
q = []
q.append(s)
size = 0
level = 0
for i in range(e):
    u, v = map(int, input().split())
    graph[u].append(v)
    graph[v].append(u)

visited = [False]*(max(graph)+1)
visited[s] = True
while q:
    print("LEVEL =", level)
    size = len(q)
    while(size):
        f = q.pop(0)
        print(f)
        for i in graph[f]:
            if visited[i] == False:
                visited[i] = True
                q.append(i)
        size = size-1
    level = level+1
